### PR TITLE
WE-458 Implement GetBhaRuns in Routes

### PR DESCRIPTION
**Pull request by eirinsvi**
# Request Template Witsml Explorer
Thank you for contributing to WITSML Explorer!
Before submitting this PR, please fill in the following template.

## Fixes
This pull request fixes #we-458 under #we-405

## Description
Added missing attributes in WitsmlBhaRun and CommonData.cs. Created BhaRun model in Api. Created getBhaRun in BhaRunService and BhaRunQueries, while adding the path to Routes.cs

## Type of change

* New feature (non-breaking change which adds functionality)

## Impacted Areas in Application
_List general components of the application that this PR will affect:_

* Api
* Frontend

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [x] PR is related to an issue

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [x] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Services/BhaRunService.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                TubularUidRef = witsmlBhaRun.Tubular.UidRef,
+                PlanDogLeg = (witsmlBhaRun.PlanDogLeg == null) ? null : new LengthMeasure { Uom = witsmlBhaRun.PlanDogLeg.Uom, Value = decimal.Parse(witsmlBhaRun.PlanDogLeg.Value) },
+                ActDogleg = (witsmlBhaRun.ActDogLeg == null) ? null : new LengthMeasure { Uom = witsmlBhaRun.ActDogLeg.Uom, Value = decimal.Parse(witsmlBhaRun.ActDogLeg.Value) },
+                ActDoglegMx = (witsmlBhaRun.ActDogLegMx == null) ? null : new LengthMeasure { Uom = witsmlBhaRun.ActDogLegMx.Uom, Value = decimal.Parse(witsmlBhaRun.ActDogLegMx.Value) },
+                DTimStart = string.IsNullOrEmpty(witsmlBhaRun.DTimStart) ? null : StringHelpers.ToDateTime(witsmlBhaRun.DTimStart),
+                DTimStop = string.IsNullOrEmpty(witsmlBhaRun.DTimStop) ? null : StringHelpers.ToDateTime(witsmlBhaRun.DTimStop),
+                DTimStartDrilling = string.IsNullOrEmpty(witsmlBhaRun.DTimStartDrilling) ? null : StringHelpers.ToDateTime(witsmlBhaRun.DTimStartDrilling),
+                DTimStopDrilling = string.IsNullOrEmpty(witsmlBhaRun.DTimStopDrilling) ? null : StringHelpers.ToDateTime(witsmlBhaRun.DTimStop),
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Services/BhaRunService.cs]: DTimStopDrilling = string.IsNullOrEmpty(witsmlBhaRun.DTimStopDrilling) ? null : StringHelpers.ToDateTime(witsmlBhaRun.**DTimStop**),
should be 
DTimStopDrilling = string.IsNullOrEmpty(witsmlBhaRun.DTimStopDrilling) ? null : StringHelpers.ToDateTime(witsmlBhaRun.**DTimStopDrilling**),

File: Src/WitsmlExplorer.Api/Services/BhaRunService.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            return new BhaRun
+            {
+                Uid = witsmlBhaRun.Uid,
+                Name = witsmlBhaRun.Name,
+                WellUid = witsmlBhaRun.WellUid,
+                WellName = witsmlBhaRun.WellName,
+                WellboreName = witsmlBhaRun.WellboreName,
+                WellboreUid = witsmlBhaRun.WellboreUid,
+                NumStringRun = witsmlBhaRun.NumStringRun,
+                StatusBha = witsmlBhaRun.NumStringRun,
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Services/BhaRunService.cs]: should be StatusBha = witsmlBhaRun.StatusBha ,

File: Src/WitsmlExplorer.Api/Routes.cs [hunk 3]
```diff
@@ -67,6 +70,7 @@ public class Routes : CarterModule
             Get("/api/wells", GetAllWells);
             Get("/api/wells/{wellUid}", GetWell);
             Get("/api/wells/{wellUid}/wellbores/{wellboreUid}", GetWellbore);
+            Get("/api/wells/{wellUid}/wellbores/{wellboreUid}/bharuns", GetBhaRun);
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Routes.cs]: The route does not include a "bhaRunUid" but it calls a method that requires it. The object view needs a route that gets all bhaRuns for a wellbore and this is missing in both bhaRunService and bhaRunQueries.

File: Src/WitsmlExplorer.Api/Query/BhaRunQueries.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+using Witsml.Data;
+using Witsml.Data.Measures;
+using Witsml.Extensions;
+using WitsmlExplorer.Api.Models;
+
+namespace WitsmlExplorer.Api.Query
+{
+    public static class BhaRunQueries
+    {
+        public static WitsmlBhaRuns GetWitsmlBhaRunByUid(string wellUid, string wellboreUid, string bhaRunUid)
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Query/BhaRunQueries.cs:12]: Ref comment in Routes, BhaRunQueries needs a query that does not include a bhaRunUid

File: Src/WitsmlExplorer.Api/Services/BhaRunService.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+namespace WitsmlExplorer.Api.Services
+{
+    public interface IBhaRunService
+    {
+        Task<BhaRun> GetBhaRun(string wellUid, string wellboreUid, string BhaRunUid);
+    }
+
+    public class BhaRunService : WitsmlService, IBhaRunService
+    {
+        public BhaRunService(IWitsmlClientProvider witsmlClientProvider) : base(witsmlClientProvider) { }
```
Comment (janburak) [inline Src/WitsmlExplorer.Api/Services/BhaRunService.cs:20]: Ref comment in Routes, BhaRunService needs a method that retrieves all bhaRuns for a given wellbore

**Review by stokkoff — APPROVED** [review]

**Pull request merged**